determined-ai/determined
# feat: add log retention for database logger

## Description


Adds log retention for database logger.

## Test Plan


Automated tests added.

Manual testing:
1. Deploy / start a cluster with this change.
2. Deploy an hpsearch experiment configured with a log retention of 0 days. 
3. Run the log cleanup command to ensure the logs are removed.
4. Modify the server configuration to use a short duration or cron expression for log cleanup.
5. Restart the server.
6. Start another hpsearch experiment with default configuration.
7. Ensure logs are retained between cleanup cycles.
8. Update experiment to have log retention of 0 days.
9. Ensure that logs are removed on next cleanup cycle.

## Commentary (optional)



## Checklist

  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.

## Ticket

DET-9995
